src/classes/UTIL_HtmlOutput_CTRL.cls

```diff
@@ -0,0 +1,135 @@
+/*
+    Copyright (c) 2018, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description The UTIL_HtmlOutput_CTRL is used to safely unescape any whitelisted tags and urls
+*/
+public with sharing class UTIL_HtmlOutput_CTRL {
+    /** @description The map of allowed tags and their temporary substitution values */
+    private static final Map<String, String> SUBSTITUTION_BY_ALLOWED_TAG = new Map<String, String> {
+        '<br>' => '|badBreak|',
+        '<br/>' => '|break|',
+        '<b>' => '|startBold|',
+        '</b>' => '|endBold|',
+        '<ol>' => '|orderList|',
+        '</ol>' => '|endOrderList|',
+        '<ul>' => '|unorderList|',
+        '</ul>' => '|endunOrderList|',
+        '<li>' => '|lineItem|',
+        '</li>' => '|endLineItem|'
```

Comment:
I think this should include heading elements (`<h1>`, etc.) and paragraph (`<p>`, `</p>`)

Reply:
I can add them, we aren't using them anywhere so I thought we should stick to the bare minimum. Since it's an added iteration for each one we add. Let me know if you would still prefer to add all tags we want to allow. I can see both sides.

Reply:
If you're building out this class, I think it's more efficient to build it out once with support for everything (or at least as much as we know is coming) than to modify this class each time we get to a new work item.